### DISCUSSION TITLE
How to use titiler ?

### DISCUSSION THREAD

**Discussion by StringKe** [developmentseed/titiler#618] [Q&A] [answered]
1. I have a 4000x4000 size png image
2. I need to move it to start at a certain latitude and longitude (leftTop) and end at a longitude and latitude (rightBottom)

Can I use titler to do this?

I don't have any knowledge about maps and I need help with this.

I'm trying to convert my image to a tiff image and submit it to `/cog/tiles/{TileMatrixSetId}/{z}/{x}/{y}. {format}` either PNG or TIFF always returns the error `CRS is invalid: None`.

**Comment by vincentsarago** [comment, marked as answer]
you first need to add `geo location` metadata to your data (e.g CRS and bounds to your file) 


```bash
gdal_translate -of COG my.png my.tif -a_srs EPSG:4326 -a_ullr <ulx> <uly> <lrx> <lry>` 
```


if you just want to visualize your image (without geo localtion) you can maybe check https://github.com/developmentseed/titiler-image

**Reply by StringKe** [reply]
Thank you for your reply!

I'm trying to generate using the following command, the image displayed in `cog/map` overlay is black, how should I keep it in color.

```shell
gdal_translate -of GTiff -a_srs EPSG:4326 -a_ullr <ulx> <uly> <lrx> <lry> -co "ALPHA=YES" -co "COMPRESS=DEFLATE" -co "PREDICTOR=2" input.png output.tif
```

When I modify `colormap_name` these images will have some color change, what should I do if I want to keep the original color?

**Reply by StringKe** [reply]
Is bix some parameters affect, when I only have a few parameters url zoom these, the color is normal, thank you very much for the program.

**Reply by StringKe** [reply]
How can I generate all the images in advance and package them?

**Reply by vincentsarago** [reply]
> When I modify colormap_name these images will have some color change, what should I do if I want to keep the original color?

@StringKe so I guess your input PNG is a palette png (mean with internal colormap).

Could you share the output of gdalinfo for both the png and the tif? 

> Is bix some parameters affect, when I only have a few parameters url zoom these, the color is normal, thank you very much for the program

I'm not sure I get that.

> How can I generate all the images in advance and package them?

please checkout https://developmentseed.org/titiler/dynamic_tiling/ the idea of TiTiler is to not create tiles in advance. if you want that checkout gdal2xyz.py script and other static tile server

**Comment by StringKe** [comment]
@vincentsarago I have a question about non-real world maps( flat game map ).

I have a map that is 8000px*8000px in size.
As the game is updated, this map size expands erratically and the scale changes.
How do I keep them in the same coordinate system.

Version1: 8000px*8000px 00001.png
Version2: 12000px*9000px 00002.png

In Version1 a lot of points may have been added such as -20,-11 for an NPC.

How can I make sure that they are the same in Version2?

**Reply by vincentsarago** [reply]
@StringKe are you using titiler-image for this? 

what does NPC mean? 

The map expand on the 4 direction? is the center of the map the origin? 

does scale means pixel resolution ?

**Reply by StringKe** [reply]
> are you using titiler-image for this?

I'm not quite sure how I should use titiler-image in leafletjs.

I'm not sure if titiler-image fixes the center point across multiple versions, or if multiple versions of the map keep the coordinate system consistent.

> what does NPC mean?

fixed point or area on the map

> The map expand on the 4 direction? is the center of the map the origin?

Yes the map will expand in all four directions.
The center point I can fix and I also know where the old center point will be after the new expansion (pixel position).

> does scale means pixel resolution ?

Probably, I'm not sure. 

-------
#### background

I'm developing a map for a game that will expand as the game is updated, and the map is designed using an image editing tool like Photoshop, not a professional GIS or anything like that.

I'm currently experiencing that the first version of my map had some points marked on it, and in the second version the points are all misaligned and don't match up.

The second version contains all the content of the first version.

**Reply by vincentsarago** [reply]
So you're saying that you are using titiler with non-geo files, which shouldn't be supported 🤷 so that's why I mentioned https://github.com/developmentseed/titiler-image

> I'm not sure if titiler-image fixes the center point across multiple versions, or if multiple versions of the map keep the coordinate system consistent. 

it doesn't 😅, it will always consider the top left of the file. 

To be honest I'm not quite sure how to do what you're trying to do, but it will pretty nice if you use titiler for this (event if titiler might be over-engineered  for this)

**Reply by StringKe** [reply]
> it doesn't 😅, it will always consider the top left of the file.

This one is my problem, I don't want to modify the data that I have saved all the points because he is very huge.

If I don't fix the data, then all the points recorded on the map in the first version will be misplaced.

![CleanShot 2024-11-18 at 01 05 18@2x](https://github.com/user-attachments/assets/3a71a34a-3cc8-4902-8419-8efbc977d0db)

The red dot in Version 1 is at 100,100.
The red dot in Version 2 are at 300,380.

Need 100,100 to be the red point in Version 2 instead of 300,380. 
I need my coordinate system to be calibrated somehow.
